subsys/canbus/isotp/isotp.c

```diff
@@ -119,7 +120,8 @@ static inline uint32_t receive_get_sf_length(struct net_buf *buf)
 static void receive_send_fc(struct isotp_recv_ctx *ctx, uint8_t fs)
 {
 	struct can_frame frame = {
-		.flags = ctx->tx_addr.ide != 0 ? CAN_FRAME_IDE : 0,
+		.flags = (ctx->tx_addr.ide != 0 ? CAN_FRAME_IDE : 0) |
+			 (IS_ENABLED(ISOTP_USE_CAN_FD) ? CAN_FRAME_FDF | CAN_FRAME_BRS : 0),
```

Comment:
Should brs (data bitrate switch) be an independent setting dependent on CAN-FD being enabled? This requires the databit rate baud and sampling to be set.

Reply:
Good idea. Perhaps we would be better off using a bitmask of options?

Reply:
Good point. I've added the `.brs` as as another parameter to the `struct isotp_msg_id`.

I decided against using the existing flags like `CAN_FRAME_FDF` and `CAN_FRAME_BRS`, as we need to apply the configuration both to the filter and the frames. The CAN driver uses different flags for frame and filter, so it would be a bit confusing.

Reply:
Ok, that makes sense and I double-checked if we could introduce the flags as part of this PR. However, it would require changes to code unrelated to CAN-FD, as we are currently using two other parameters for the ISO-TP extended addressing and fixed addressing features.

For this reason, I suggest to switch to flags in a follow-up PR.

Reply:
Change of mind. Let's get the flags in first: #62255

I'll get back to this PR afterwards.